src/driver.rs

```diff
@@ -466,4 +501,15 @@ mod tests {
         assert!(DriverManager::count() > 0);
         assert!(DriverManager::get_driver(0).is_ok());
     }
+
+    #[test]
+    fn test_driver_iterator() {
+        assert_eq!(DriverManager::count(), DriverIterator::new().count());
+        assert_eq!(DriverManager::count(), DriverManager::all().count());
+
+        let drivers: HashSet<String> = DriverManager::all().map(|d| d.short_name()).collect();
```

Comment:
This feels a little overkill, I'd just keep the `count` test.

Reply:
I was thinking the same in the beginning, but repeating the same driver `DriverManager::count()` times would pass the test. Although that's a very unlikely mistake, this feels like a complete test.